vortex-sampling-compressor/src/compressors/fsst.rs

```diff
@@ -0,0 +1,65 @@
+use std::collections::HashSet;
+
+use vortex::array::{VarBinArray, VarBinViewArray};
+use vortex::encoding::EncodingRef;
+use vortex::{ArrayDType, ArrayDef, IntoArray};
+use vortex_dict::DictArray;
+use vortex_dtype::DType;
+use vortex_error::{vortex_bail, VortexResult};
+use vortex_fsst::{fsst_compress, FSSTEncoding, FSST};
+
+use super::{CompressedArray, CompressionTree, EncodingCompressor};
+use crate::SamplingCompressor;
+
+#[derive(Debug)]
+pub struct FSSTCompressor;
+
+impl EncodingCompressor for FSSTCompressor {
+    fn id(&self) -> &str {
+        FSST::ID.as_ref()
+    }
+
+    fn can_compress(&self, array: &vortex::Array) -> Option<&dyn EncodingCompressor> {
+        // FSST arrays must have DType::Utf8.
+        //
+        // Note that while it can accept binary data, it is unlikely to perform well.
+        if !matches!(array.dtype(), &DType::Utf8(_)) {
+            return None;
+        }
+
+        // FSST cannot be applied recursively.
+        if array.encoding().id() == FSST::ID {
+            return None;
+        }
+
+        Some(self)
+    }
+
+    fn compress<'a>(
```

Comment:
I think you want to recurse and try to compress the codes?

Reply:
At least in BtrBlocks, they treated FSST as a terminal node for recursion


![image](https://github.com/user-attachments/assets/12f5b704-1d6f-4a72-b7b0-0ee37f71db16)

The output of FSST is an array of 8 byte symbols and a varbin array of code IDs. The only thing you could really do to the varbin is dictionary compress it, and I think we want to only ever do dict-then-fsst instead of fsst-then-dict

Reply:
(The green nodes are recursion points)